### PR TITLE
(PCP-523) More unit and integration tests

### PR DESCRIPTION
**Pull request by parisiale**

### CODE REVIEW

**Comment by parisiale** [comment]
More integration tests to come, don't merge.

File: test/integration/puppetlabs/pcp/broker/service_test.clj [hunk 1]
```diff
@@ -192,34 +215,34 @@
...
-(defn is-error-message
-  "Assert that the message is a PCP error message with the specified description"
-  [message version expected-description]
-  (is (= "http://puppetlabs.com/error_message" (:message_type message)))
-  (if (= "v1.0" version)
-    (is (= nil (:in-reply-to message)))
-    (is (:in-reply-to message)))
-  (is (= expected-description (:description (message/get-json-data message)))))
+(deftest reply-to-malformed-messages-during-association-test
+  (testing "During association, broker replies with error_message to a deserialization failure"
```
Comment (MikaelSmith) [inline test/integration/puppetlabs/pcp/broker/service_test.clj]: @er0ck this seems along the lines of some things you were trying to test with ruby-pcp-client

File: test/unit/puppetlabs/pcp/broker/core_test.clj [hunk 2]
```diff
@@ -464,7 +464,34 @@
...
+                  (fn [websocket raw-message] (reset! error-msg (message/decode raw-message)))]
+      (testing "sends errror_message correctly and returns nil if received msg is NOT given"
+        (let [received-msg nil
+              error-description "something wrong!"
+              outcome (send-error-message received-msg error-description connection)
+              msg-data (message/get-json-data @error-msg)]
+          (is (nil? outcome))
+          (is (not (contains? msg-data :id)))
+          (is (= error-description (:description msg-data)))))
+      (testing "sends errror_message correctly and returns nil if received msg is given"
```
Comment (MikaelSmith) [inline test/unit/puppetlabs/pcp/broker/core_test.clj]: typo in errror

**Comment by MikaelSmith** [comment]
Other than the typo, 👍

**Comment by parisiale** [comment]
Updated with the typo fix.
Skipping CI as Travis is precious.

**Pull request merged**